src/aliases/store/store.go

```diff
@@ -0,0 +1,50 @@
+package aliasstore
+
+import "context"
+
+//go:generate mockgen -source=store.go -destination=mock/store.go -package=mock
+
+// Store handles the alias storing.
+type Store interface {
+	// CreateAlias creates an alias in the registry.
+	CreateAlias(ctx context.Context, registry RegistryName, alias Alias) error
+	// GetAlias gets an alias from the registry.
+	GetAlias(ctx context.Context, registry RegistryName, aliasKey AliasKey) (*Alias, error)
+	// UpdateAlias updates an alias in the registry.
+	UpdateAlias(ctx context.Context, registry RegistryName, alias Alias) error
+	// GetAlias deletes an alias from the registry.
+	DeleteAlias(ctx context.Context, registry RegistryName, aliasKey AliasKey) error
+
+	// ListAlias lists all aliases from a registry.
+	ListAliases(ctx context.Context, registry RegistryName) ([]Alias, error)
+
+	// DeleteRegistry deletes a registry, with all the aliases it contained.
+	DeleteRegistry(ctx context.Context, registry RegistryName) error
+}
+
+// Alias allows the user to associates a RegistryName + a Key to 1 or more public keys stored
+// in Value. The Value has 2 formats:
+// - a JSON string if AliasKind is an AliasKindString.
+// - a JSON array of strings if AliasKind is an AliasKindArray.
+type Alias struct {
+	tableName struct{} `pg:"aliases"` // nolint:unused,structcheck // reason
+
+	Key          AliasKey     `pg:",pk"`
+	RegistryName RegistryName `pg:",pk"`
```

Comment:
From what I see, it seems you have 2 entities in your data representation:
- A `Registry`, which is a list of aliases, a registry has a `name`
- An `Alias`, which is a resource that gets added to a registry.

So apparently you need 2 models, one `Registry` and one `Alias` where the alias is linked to a registry by name. Having a unique index on `(alias, registry)` is also fine but inferior representation of the data. 

PS: You don't seem to be creating registries so maybe I'm wrong on this one and the registry is something that we do not control?

Reply:
We control the registry, it is just not used in the DB as it is because we don't really need to. It is just to scope aliases from other registries.
I'll split the entities and models then.